base/twiceprecision.jl

```diff
@@ -196,8 +196,21 @@ ranges, you can set `nb = ceil(Int, log2(len-1))`.
 struct TwicePrecision{T}
     hi::T    # most significant bits
     lo::T    # least significant bits
+
+    function TwicePrecision{T}(hi::T, lo::T) where {T}
+        isconcretetype(T) || error("$T isn't a concrete type")
+        (T <: Integer) &&
+            error("TwicePrecision{<:Integer} is nonsensical")
```

Comment:
This is not a good way to communicate that requirement - it doesn't tell a user anything about how they might fix the issue if they encounter the error message. Something like

```
throw(ArgumentError("Cannot construct a `TwicePrecision{<:Integer}` - only floating point types and quantities of them are supported."))
```

is more appropriate (though it does feel a bit weird to call out "quantities", since we don't really have any like that in Base itself... Maybe that's an argument that Quantities.jl should not rely on TwicePrecision).